src/linker/Linker/TypeNameResolver.cs

```diff
@@ -98,7 +102,7 @@ TypeReference ResolveTypeName (AssemblyDefinition assembly, TypeName typeName)
 				if (genericTypeRef == null)
 					return null;
 
-				TypeDefinition genericType = _context.TryResolve (genericTypeRef);
+				TypeDefinition? genericType = _context.TryResolve (genericTypeRef);
```

Comment:
Should we check for null here?

Reply:
Good catch - this is the kind of case that unfortunately won't be caught by the analysis. Looking at it I think the `TryResolve` call is actually unnecessary since resolving the generic type above should always return a TypeDefinition, so I replaced it with an assert. PTAL @tlakollo and @mateoatr